hooks/products/useProductDetail.js

```diff
@@ -0,0 +1,125 @@
+import { isWidthUp } from "@material-ui/core/withWidth";
+import Router from "translations/i18nRouter";
+import priceByCurrencyCode from "lib/utils/priceByCurrencyCode";
+import variantById from "lib/utils/variantById";
+import useStores from "hooks/useStores";
+import useWidth from "hooks/useWidth";
+import useProductViewedEvent from "hooks/analytics/useProductViewedEvent";
+
+/**
+ * Hook to process product details
+ *
+ * @returns {Object} the product details functions
+ */
+export default function useProductDetail(props) {
+  const width = useWidth();
+  const { uiStore, routingStore } = useStores();
+  const trackProductViewedEvent = useProductViewedEvent();
+  const { openCartWithTimeout, pdpSelectedOptionId, pdpSelectedVariantId } = uiStore;
+
+  const {
+    addItemsToCart,
+    currencyCode,
+    product
+  } = props;
+
+  const selectVariant = (variant, optionId) => {
+    // Select the variant, and if it has options, the first option
+    const variantId = variant._id;
+    let selectOptionId = optionId;
+    if (!selectOptionId && variant.options && variant.options.length) {
+      selectOptionId = variant.options[0]._id;
+    }
```

Comment:
Could this be simplified in a manner similar to:

```
const selectedVariantId = variant._id;
const firstOption = [variant.options];
const variantOrOptionId = optionId || firstOption?._id || selectedVariantId
```

Then used to remove the boolean logic from the `url` assignment?

Reply:
Thanks for the work you put into refactoring this @jhonsfran1165 !
I do agree that instead of all the separate calls to determine the variants in several places, we could just determine and memoize them at the top of the component.

However I think that doesn't necessarily needs to be part of this PR, as this is complexity which also was there before.